stetho/src/main/java/com/facebook/stetho/inspector/elements/android/AccessibilityNodeInfoWrapper.java

```diff
@@ -100,4 +108,169 @@ public static String getIgnoredReasons(AccessibilityNodeInfoCompat node, View vi
     return "View is not actionable and has no description.";
   }
 
+  @Nullable
+  public static String getFocusableReasons(AccessibilityNodeInfoCompat node, View view) {
+    boolean hasText = AccessibilityUtil.hasText(node);
+    boolean isCheckable = node.isCheckable();
+    boolean hasNonActionableSpeakingDescendants =
+        AccessibilityUtil.hasNonActionableSpeakingDescendants(node, view);
+
+    if (AccessibilityUtil.isActionableForAccessibility(node)) {
+      if (node.getChildCount() <= 0) {
+        return "View is actionable and has no children.";
+      } else if (hasText) {
+        return "View is actionable and has a description.";
+      } else if (isCheckable) {
+        return "View is actionable and checkable.";
+      } else if (hasNonActionableSpeakingDescendants) {
+        return "View is actionable and has non-actionable descendants with descriptions.";
+      }
+    }
+
+    if (AccessibilityUtil.isTopLevelScrollItem(node, view)) {
+      if (hasText) {
+        return "View is a direct child of a scrollable container and has a description.";
+      } else if (isCheckable) {
+        return "View is a direct child of a scrollable container and is checkable.";
+      } else if (hasNonActionableSpeakingDescendants) {
+        return
+            "View is a direct child of a scrollable container and has non-actionable " +
+            "descendants with descriptions.";
+      }
+    }
+
+    if (hasText) {
+      return "View has a description and is not actionable, but has no actionable ancestor.";
+    }
+
+    return null;
+  }
+
+  @Nullable
+  public static String getActions(AccessibilityNodeInfoCompat node) {
+    final StringBuilder actionLabels = new StringBuilder();
+    final String separator = ", ";
+
+    for (AccessibilityActionCompat action : node.getActionList()) {
+      if (actionLabels.length() > 0) {
```

Comment:
this seems like the wrong logic for adding the separator. don't you want to append it for everything except the 1st item? I think this code will result in a string like: `, first, second, third` because if `actionLabels.length() > 0` then you'll still append the separator on the first item